### DISCUSSION TITLE
CDT Ubuntu gcc -lm not fixing undefined reference to pow()

### DISCUSSION THREAD

**Discussion by gregwillits** [eclipse-cdt/cdt#628] [Q&A] [answered]
According to 4,231,864 posts I've read, using pow() for Linux requires that `-lm` be added to the gcc command. (Eclipse/gcc on macOS doesn't need this, so now compiling on Linux, this is the first time I've run into it.)

Apparently after reading said posts, I can't figure out where to add that. I've seen posts explicitly saying to add it to the linker command (logical) as shown in my attached. And I've tried a variety of other ways to add it within the GUI as well.

I've tested this with CDT 4.29 running on Ubuntu 20 / gcc 9..4.0 / make 4.2.1 and Ubuntu 22 / gcc 11.4.0 / make 4.3. 

And it doesn't matter if I use the CDT Internal Builder, or use the Gnu Make Builder. Same error.

I must be missing something obvious to everyone else.

Ideas? Thanks for your time.

<img width="1263" alt="pow-test" src="https://github.com/eclipse-cdt/cdt/assets/248531/c1528e6e-f2a6-4380-a025-7cd7d88ebfab">

**Comment by jld01** [comment, marked as answer]
Try adding the single character "m" (without the quotes) to the list of libraries on the Libraries page of Tool Settings.

**Reply by gregwillits** [reply]
Thought I tried that as one of my permutations, but this time it worked. After doing so, the `-lm` appears in the `All options` field.

Thank you.
